sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -24,20 +24,56 @@ class ResourceOptionsManager(
    * Static variables and methods.
    */
   companion object {
-    private lateinit var default: ResourceOptionsManager
+    private var default: ResourceOptionsManager? = null
+    internal fun getTokenResId(context: Context): Int = context.resources.getIdentifier(
+      MAPBOX_ACCESS_TOKEN_RESOURCE_NAME,
+      "string",
+      context.packageName
+    )
+
+    /**
+     * Convenience function to remove the default instance. Calling `getDefault` again will re-create the default instance.
+     */
+    fun destroyDefault() {
+      default = null
+    }
```

Comment:
@julianrex here we made this static `destroyDefault` function public to users, and removed the `reset` method on the `ResourceOptionsManager`, let's discuss and align it.

Reply:
Addressed in https://github.com/mapbox/mapbox-maps-ios/pull/406